moor/test/expressions/subquery_test.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:test/test.dart';
+import 'package:moor/moor.dart';
+
+import '../data/tables/todos.dart';
+
+void main() {
+  test('exists expressions are generated', () {
+    final db = TodoDb();
+    final subquery = db.select(db.users)
+      ..where((tbl) => tbl.isAwesome.equals(true));
+    final existsExpression = existsQuery(subquery);
+
+    final context = GenerationContext.fromDb(db);
+    existsExpression.writeInto(context);
+
+    expect(context.sql, 'EXISTS (SELECT * FROM users WHERE is_awesome = ?)');
+    expect(context.boundVariables, [1]);
```

Comment:
I have a neat test utility that checks all this at once:

```dart
  expect(existsExpression, generates('EXISTS (SELECT * FROM users WHERE is_awesome = ?)', [1]));
```

You'd have to import `../data/expect_generated.dart`.